--- Android.mk
+++ Android.mk
@@ -2,11 +2,12 @@
 
 include $(CLEAR_VARS)
 
-LOCAL_MODULE := SDL2_ttf
+LOCAL_MODULE := SDL2_ttf_static
+LOCAL_MODULE_FILENAME := libSDL2_ttf
 
 FREETYPE_LIBRARY_PATH := external/freetype-2.4.12
 
-LOCAL_C_INCLUDES := $(LOCAL_PATH)
+LOCAL_C_INCLUDES := $(LOCAL_PATH) ../SDL2-2.0.3/include
 
 LOCAL_SRC_FILES := SDL_ttf.c \
 
@@ -59,8 +60,8 @@
         $(FREETYPE_LIBRARY_PATH)/src/winfonts/winfnt.c
 endif
 
-LOCAL_SHARED_LIBRARIES := SDL2
+LOCAL_SHARED_LIBRARIES :=
 
 LOCAL_EXPORT_C_INCLUDES += $(LOCAL_C_INCLUDES)
 
-include $(BUILD_SHARED_LIBRARY)
+include $(BUILD_STATIC_LIBRARY)
